### ISSUE TITLE
Memory/CPU Limits

### ISSUE THREAD

**Issue by longshorej** [lightbend/reactive-cli#71]
If the annotations contain memory or cpu values, we need to set the appropriate values in the generated pod controller.

Also, investigate diskSpace which is present in the annotations but may or may not have Kubernetes semantics. It will for Marathon, however.

**Comment by longshorej** [comment]
* As part of this, I'd like to rename `nrOfCpus` to `cpu`.
* We should set request and limit to the same value for both memory and cpu
* The values can be specified directly as-is. Kubernetes and Mesos both use the same concept for limits, i.e. it's a fraction of 1024 (default cgroup shares value)
* In the future, we can introduce a ratio setting that defaults to 1, and maps to marathon/mesos if appropriate

These are the settings for the pods:

```
spec.containers[].resources.limits.cpu
spec.containers[].resources.limits.memory
spec.containers[].resources.requests.cpu
spec.containers[].resources.requests.memory
```

More info: https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container

**Comment by mitkus** [comment]
My PR is here: https://github.com/lightbend/reactive-cli/pull/72

Storage limiting is an alpha feature in Kubernetes 1.8 (https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/#local-ephemeral-storage-alpha-feature). I didn't add storage limit support yet, but it could easily be done in separate PR.

It seems cpu values are measured in "cpu units" where 1 corresponds to a single cpu core. Docker expects fractions of 1024, but this conversion is done by Kubernetes.

**Comment by longshorej** [comment]
Fixed via #72
